lib/private/Search/Filter/TypeDateTime.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright 2023 Benjamin Gaussorgues <benjamin.gaussorgues@nextcloud.com>
+ *
+ * @author Benjamin Gaussorgues <benjamin.gaussorgues@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OC\Search\Filter;
+
+use DateTimeImmutable;
+use OC\Search\Filter;
+
+class TypeDateTime extends Filter {
+	protected function parse(string $value): DateTimeImmutable {
+		if (filter_var($value, FILTER_VALIDATE_INT)) {
+			$value = '@'.$value;
+		}
+
+		return new DateTimeImmutable($value);
+	}
+
+	public function get(): DateTimeImmutable {
+		return $this->value;
+	}
```

Comment:
```suggestion
	}
	
	public function getSearchComparison(): ISearchComparison {
		return new SearchComparison(ISearchComparison::COMPARE_GREATER_THAN_EQUAL, $this->name, $this->value);
	}
```

And either you have two classes for >= and <=, or you add a property to store the operator?